packages/tenants-nodejs/src/plugin.ts

```diff
@@ -0,0 +1,1064 @@
+                },
+              },
+              handler: withRequestHandler(async (req, res, session, userContext) => {
+                if (!session) {
+                  throw new Error("Session not found");
+                }
+
+                const userDetails = await supertokens.getUser(session.getUserId());
+                if (!userDetails) {
+                  return {
+                    status: "USER_NOT_FOUND",
+                    message: "User not found",
+                  };
+                }
+
+                const tenantId = session.getTenantId();
+
+                const tenantDetails = await MultiTenancy.getTenant(tenantId);
+                if (!tenantDetails) {
+                  return {
+                    status: "ERROR",
+                    message: "Tenant not found",
+                  };
+                }
+
+                // Associate the user with the tenant
+                // NOTE: Unlikely that login method will be undefined so we can ignore that.
+                await MultiTenancy.disassociateUserFromTenant(tenantId, userDetails.loginMethods[0]!.recipeUserId);
+
+                // Do session.revoke and create a new session instead of the above
+                await session.revokeSession();
+                await Session.createNewSession(
+                  req,
+                  res,
+                  tenantId,
+                  session.getRecipeUserId(),
+                  session.getAccessTokenPayload(),
+                  undefined,
+                  userContext,
+                );
+
+                return {
+                  status: "OK",
+                  message: "User disassociated from tenant",
+                };
+              }),
+            },
+            {
+              path: `${HANDLE_BASE_PATH}/users`,
+              method: "post",
+              verifySessionOptions: {
+                sessionRequired: true,
+                overrideGlobalClaimValidators: (globalValidators) => {
+                  return [
+                    ...globalValidators,
+                    UserRoles.UserRoleClaim.validators.includesAny([ROLES.ADMIN, ROLES.MEMBER]),
+                  ];
+                },
+              },
+              handler: withRequestHandler(async (req, res, session) => {
+                if (!session) {
+                  throw new Error("Session not found");
+                }
+
+                const tenantIdToUse = session.getTenantId();
+                const getUsersResponse = await implementation.getTenantUsers(tenantIdToUse);
+
+                return getUsersResponse;
+              }),
+            },
+            {
+              path: `${HANDLE_BASE_PATH}/remove`,
+              method: "post",
+              verifySessionOptions: {
+                sessionRequired: true,
+                overrideGlobalClaimValidators: (globalValidators) => {
+                  return [...globalValidators, UserRoles.UserRoleClaim.validators.includesAny([ROLES.ADMIN])];
```

Comment:
This kind of conflicts with the `canRemoveUserFromTenant` check below.

Reply:
I think if we go to a permission-based model, it will work out OK. In that case, the primary use case for `canRemoveUserFromTenant` may be to validate that the specific target user can be removed or not from the tenant (instead of validating that the session user can do the action)